numba/tests/test_nrt.py

```diff
@@ -546,5 +552,106 @@ def foo(x):
         self.assertEqual(foo(10), 22) # expect (10 + 1) * 2 = 22
 
 
+@unittest.skipUnless(cffi_support.SUPPORTED, "cffi required")
+class TestNrtExternalCFFI(MemoryLeakMixin, TestCase):
+    """Testing the use of externally compiled C code that use NRT
+    """
+
+    def compile_cffi_module(self, name, source, cdef):
+        from cffi import FFI
+
+        ffi = FFI()
+        ffi.set_source(name, source, include_dirs=[include_path()])
+        ffi.cdef(cdef)
+        tmpdir = temp_directory("cffi_test_{}".format(name))
+        ffi.compile(tmpdir=tmpdir)
+        sys.path.append(tmpdir)
+        try:
+            mod = import_dynamic(name)
+        finally:
+            sys.path.remove(tmpdir)
+
+        return ffi, mod
+
+    def get_nrt_api_table(self):
+        from cffi import FFI
+
+        ffi = FFI()
+        nrt_get_api = ffi.cast("void* (*)()", _nrt_python.c_helpers['get_api'])
+        table = nrt_get_api()
+        return table
+
+    def test_manage_memory(self):
+        name = "{}_test_manage_memory".format(self.__class__.__name__)
+        source = r"""
+#include <stdio.h>
+#include "numba/runtime/nrt_external.h"
+
+void my_dtor(void *ptr) {
+    free(ptr);
+}
+
+NRT_MemInfo* test_nrt_api(NRT_api_functions *nrt) {
+    void * data = malloc(10);
+    NRT_MemInfo *mi = nrt->manage_memory(data, my_dtor);
```

Comment:
it will be called as this test function is being returned; i.e. `mi` go out of scope. but i should add an other test to demonstrate that the dtor is indeed called by making the dtor do something more easily observable.